consensus/spos/worker.go

```diff
@@ -711,6 +728,48 @@ func (wrk *Worker) ResetConsensusMessages() {
 	wrk.consensusMessageValidator.resetConsensusMessages()
 }
 
+func (wrk *Worker) processEquivalentMessage(msgType consensus.MessageType, blockHeaderHash []byte) error {
+	// early exit if the message is not with final info
+	if !wrk.consensusService.IsMessageWithFinalInfo(msgType) {
+		return nil
+	}
+
+	hdrHash := string(blockHeaderHash)
+
+	wrk.mutEquivalentMessages.Lock()
+	defer wrk.mutEquivalentMessages.Unlock()
+
+	// if an equivalent message was seen before, return error to stop further broadcasts
```

Comment:
slightly optimization for L743 - L49
```GO
numMessages := wrk.equivalentMessages[hdrHash]
wrk.equivalentMessages[hdrHash] = numMessages+1
if numMessages > 0 {
    return ErrEquivalentMessageAlreadyReceived
}
```
as this one does one read & one write on the map. The original code does 2 reads & one write

Reply:
applied